### DISCUSSION TITLE
How to include parameters from MSBuild or Build Environment Variables

### DISCUSSION THREAD

**Discussion by jonmotos** [oleg-shilo/wixsharp#1686] [Q&A]
Is there an example illustrating how to use environment variables from the *build environment itself*? For example if I have a ProductVersion variable.
I would like to accomplish two things:
1. build the MSI to a specific x64 ($(Platform)) and Debug ($(Configuration)) directory,
2. set a wix variable for example named "ProductVersion" to be used in the wix authoring process. This way this would get utilized during CI/CD.

Am I missing something simple here? Or is it unusual to need these variables internal to the wix steps?

Thanks in advance!

**Comment by oleg-shilo** [comment]
MSBuild variables are required for customizing the build actions expressed as XML. This automation architecture is limited and does not support any other way of customization.

With WixSharp the build automation is implemented as a build script of the C#syntax. Thus you have the whole power of the full-scale programming language. You can:

- analyse OS and process environment (e.g env vas)
- analyze the file system
- define conditional behaviour based on the [custom symbols](https://learn.microsoft.com/en-us/dotnet/csharp/language-reference/compiler-options/language#defineconstants)
- analyze the build script assembly itself with Reflection (e.g. get version) 
- analyze command line params (e.g. you can pass your build-envars as CLI args. Remember, when the build script is executed it is simply invoked from as a targetexe from the post-build event
